### PR TITLE
Y2K-1795 Delete account on mobile

### PR DESCRIPTION
**Pull request by Daniel-Wang**
The approach I took was adding an optional passphrase param to the RPC and prompting the user for a password in the `mobile with password` case.
The rest of the cases remain unchanged and work for me from local testing

@keybase/y2ksquad

### CODE REVIEW

File: shared/reducers/settings.tsx [hunk 1]
```diff
@@ -169,6 +169,9 @@ const passwordActions: Container.ActionHandler<Actions, Types.State> = {
     password.error = undefined
     password.newPasswordConfirm = action.payload.password
   },
+  [SettingsGen.resetCheckPasswordIsCorrect]: draftState => {
+    draftState.checkPasswordIsCorrect = undefined
```
Comment (Daniel-Wang) [inline shared/reducers/settings.tsx:173]: Added this to reset the "checked password" state, which isn't reset to `undefined`

**Comment by zapu** [comment]
we should have a test for case where the engine gets a passphrase in argument

File: shared/actions/json/settings.json [hunk 2]
```diff
@@ -213,6 +215,7 @@
     "loadedHasRandomPw": {
       "randomPW": "boolean"
     },
+    "resetCheckPasswordIsCorrect": {},
```
Comment (jakob223) [inline shared/actions/json/settings.json:218]: ?

File: shared/settings/delete-confirm/check-passphrase.native.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    dispatch(nav.safeNavigateUpPayload())
+  }
+  const onCheckPassword = (password: string) => {
+    if (password) {
+      dispatch(SettingsGen.createCheckPassword({password: new HiddenString(password)}))
+    }
+  }
+  const deleteForever = () => dispatch(SettingsGen.createDeleteAccountForever({passphrase: password}))
+
+  const waitingKey = Container.useSelector(state => Container.anyWaiting(state, Constants.settingsWaitingKey))
```
Comment (jakob223) [inline shared/settings/delete-confirm/check-passphrase.native.tsx:29]: there's `Container.useAnyWaiting` if you like

**Comment by heronhaye** [comment]
Should confirm with Patrick this is the right way to send passwords via AVDL

**Comment by jakob223** [comment]
cc @patrickxb

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**